### DISCUSSION TITLE
Can we register Protocol by using Factory?

### DISCUSSION THREAD

**Discussion by kathiravan-raju** [hmlongco/Factory#167] [Q&A] [answered]
We have implements method in the Resolver framework to register the protocol like shown below: 
  `register { MyService() }
            .implements(MyProtocol.self, name: .sample)
            .scope(.application)`

 By using Factory, I tried the below code but didn’t work 
 ` var myServices: Factory<MyService> {
        Factory(self) {
            MyService()
        }.register {
            MyProtocol.self // Error here
        }
    } `
  
  Do we have any similar method in Factory to register a Protocol?. Can anyone please suggest an example using factory.

**Comment by hmlongco** [comment, marked as answer]
Answered in issues...
https://github.com/hmlongco/Factory/issues/22

**Reply by kathiravan-raju** [reply]
Thank you Michael
